### PR TITLE
Allow implementation of virtual interface methods in VB.

### LINKED ISSUE #35821: VB DIM: Class cannot implement a virtual interface method
```
        <Fact>
        Public Sub MethodImplementation_02()

            Dim csSource =
"
public interface I1
{
    void M1() 
    {
    }
}
"
            Dim csCompiation = GetCSharpCompiation(csSource).EmitToImageReference()

            Dim source1 =
<compilation>
    <file name="c.vb"><![CDATA[
Public Class C
    Implements I1

    Sub M1() Implements I1.M1
        System.Console.WriteLine("C.M1")
    End Sub

    Shared Sub Main()
        Dim i1 As I1 = new C()
        i1.M1()
    End Sub
End Class
]]></file>
</compilation>

            Dim comp1 = CreateCompilation(source1, options:=TestOptions.DebugExe, targetFramework:=TargetFramework.NetStandardLatest, references:={csCompiation})
            CompileAndVerify(comp1, expectedOutput:=If(ExecutionConditionUtil.IsMonoOrCoreClr, "C.M1", Nothing), verify:=VerifyOnMonoOrCoreClr)
        End Sub
```

Observed:
```
c.vb(4) : error BC30401: 'M1' cannot implement 'M1' because there is no matching sub on interface 'I1'.
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Closes #35821.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@ dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

File: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb [hunk 1]
```diff
@@ -3795,10 +3790,36 @@ End Class
...
+                                Assert.False(p1Get.IsMetadataFinal)
+                                Assert.False(p1Get.IsMustOverride)
+                                Assert.False(p1Get.IsOverridable)
+                                Assert.Empty(p1Get.ExplicitInterfaceImplementations)
+                            End Sub
+
+            CompileAndVerify(comp1, expectedOutput:=If(ExecutionConditionUtil.IsMonoOrCoreClr, "C.P1.Set", Nothing), verify:=VerifyOnMonoOrCoreClr, sourceSymbolValidator:=validator, symbolValidator:=validator)
+
+            comp1 = CreateCompilation(source1, options:=TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All), targetFramework:=TargetFramework.NetStandardLatest, references:={csCompilation})
+            CompileAndVerify(comp1, expectedOutput:=If(ExecutionConditionUtil.IsMonoOrCoreClr, "C.P1.Set", Nothing), verify:=VerifyOnMonoOrCoreClr, sourceSymbolValidator:=validator, symbolValidator:=validator)
```
Comment (RikkiGibson) [inline src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb]: >Nothing [](start = 107, length = 7)

Running this executable on .NET Framework is an unsupported scenario, right? This is just verifying the runtime behavior?

Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/DefaultInterfaceImplementationTests.vb]: > Running this executable on .NET Framework is an unsupported scenario, right? This is just verifying the runtime behavior?

Correct, ```Nothing``` ensures we verify execution only on CoreClr or Mono.

---
In reply to: [313599079](https://github.com/dotnet/roslyn/pull/37914#discussion_r313599079) [](ancestors = 313599079)

**Pull request merged**